src/utils/conversion.ts

```diff
@@ -0,0 +1,119 @@
+function extractConversation(data: any): any[] {
```

Comment:
Can you try to reduce these `any` ? Simply declare the interfaces and use them.

Reply:
I've never really worked with typescript before, so I'm not really sure how to best do this.

Reply:
can ask chatgpt 😄